tests/unit/models/addon-test.js

```diff
@@ -407,6 +407,86 @@ describe('models/addon.js', function() {
       });
     });
 
+    describe('hintingEnabled', function() {
+      var originalEnvValue, originalTestCommand, addon, project;
+
+      beforeEach(function() {
+        var MyAddon = Addon.extend({
+          name: 'test-project',
+          root: 'foo'
+        });
+
+        var projectPath = path.resolve(fixturePath, 'simple');
+        var packageContents = require(path.join(projectPath, 'package.json'));
+
+        project = new Project(projectPath, packageContents);
+
+        addon = new MyAddon(project);
+
+        originalEnvValue = process.env.EMBER_ADDON_ENV;
+        originalTestCommand = process.env.EMBER_CLI_TEST_COMMAND;
+      });
+
+      afterEach(function() {
+        addon.app = {
+          options: {}
+        };
+
+        if (originalEnvValue === undefined) {
+          delete process.env.EMBER_ADDON_ENV;
+        } else {
+          process.env.EMBER_ADDON_ENV = originalEnvValue;
+        }
+
+        if (originalTestCommand === undefined) {
+          delete process.env.EMBER_CLI_TEST_COMMAND;
+        } else {
+          process.env.EMBER_CLI_TEST_COMMAND = originalTestCommand;
+        }
+      });
+
+      it('returns true when `EMBER_ENV` is not set to production and options.hinting is true', function() {
+        process.env.EMBER_ENV = 'development';
+
+        addon.app = {
+          options: { hinting: true }
+        };
+
+        expect(addon.hintingEnabled()).to.eql(true);
+      });
+
+      it('returns true when `EMBER_CLI_TEST_COMMAND` is set and options.hinting is true', function() {
+        addon.app = {
+          options: { hinting: true }
+        };
+
+        expect(addon.hintingEnabled()).to.eql(true);
+      });
+
+      it('returns false when `EMBER_ENV` is set to production, `EMBER_CLI_TEST_COMMAND` is unset and options.hinting is true', function() {
+        process.env.EMBER_ENV = 'production';
+        delete process.env.EMBER_CLI_TEST_COMMAND;
+
+        addon.app = {
+          options: { hinting: true }
+        };
+
+        expect(addon.hintingEnabled()).to.eql(false);
```

Comment:
it would be `.ok;`, not `.ok();`. chai is using properties here instead of functions and using them as function isn't actually running any assertions. I don't agree with using properties there, but that's clearly not my decision to make 😉 

I'm fine with `.ok` here too, although if we can be more specific with `.true/false` then we probably should use that. I'm not sure if there are cases where that method can return something trueish which isn't actually `true` itself.